### PR TITLE
feat(datatype): support gis datatype for mysql and ob mysql

### LINKED ISSUE #899: [Feature]: mysql and ob mysql mode support gis datatype
### Is your feature request related to a problem?

no

### Describe the solution you'd like

mysql support gis datatype, Supports creating, viewing, and managing tables containing gis field types.
And supports processing result sets containing gis field types：

- GEOMETRY

- POINT

- LINESTRING

- POLYGON

- MULTIPOINT

- MULTILINESTRING

- MULTIPOLYGON

- GEOMETRYCOLLECTION

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PeachThinking**
#### What type of PR is this?

type-feature
module-Data viewing and editing

#### What this PR does / why we need it:
mysql and ob mysql 3.2.4 and above support gis type fields, including：

- GEOMETRY

- POINT

- LINESTRING

- POLYGON

- MULTIPOINT

- MULTILINESTRING

- MULTIPOLYGON

- GEOMETRYCOLLECTION

add `MySQLGeometryMapper` to convert gis datatype from `jdbc resultset` to `String`. In jdbc resultset all of the gis datatype‘s `ColumnTypeName` is `GEOMETRY`.
SRID is 0:
image
SRID is not 0（mysql 8.x）:
image

add `MySQLGeometryConverter`，When editing data on the front end, use the ST_GeomFromText function to construct a spatial geometry based on the input gis type WKT text. generated sql :
```
-- SRID is 0
update `jingtian_test`.`gis_table` set `point_field` = ST_GeomFromText('POINT (40.7128 -74.0)') where `id`=3;

-- SRID is not 0
update `jingtian_test`.`gis_srid` set `point_field` = ST_GeomFromText('POINT (-74.006 40.7128)', 4322) where `id`=3;
```

#### Which issue(s) this PR fixes:

Fixes #899

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public DefaultJdbcRowMapper(@NonNull ConnectionSession session) {
             mapperList.add(new MySQLDatetimeMapper());
             mapperList.add(new MySQLYearMapper());
             mapperList.add(new MySQLTimestampMapper());
+            mapperList.add(new MySQLGeometryMapper());
```
Comment (yhilmare) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java:49]: should it be used on rollback plan?

File: server/odc-core/src/test/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLGeometryMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        prepareEnv();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        destroyEnv();
+    }
+
+    @Test
+    public void test_POINT_datatype() throws Exception {
```
Comment (yhilmare) [inline server/odc-core/src/test/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLGeometryMapperTest.java:55]: we'd better put only one assert in a test case

**Review by MarkPotato777 — COMMENTED** [review]
Good job! 👍🏻

File: server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java (hunk 1 above)
Comment (PeachThinking) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/sql/execute/mapper/DefaultJdbcRowMapper.java:49]: Yes. add MySQLGeometryMapper in RollBackPlanJdbcRowMapper

File: server/odc-core/src/test/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLGeometryMapperTest.java (hunk 2 above)
Comment (PeachThinking) [reply server/odc-core/src/test/java/com/oceanbase/odc/core/sql/execute/mapper/MySQLGeometryMapperTest.java:55]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/service/dml/converter/MySQLGeometryConverter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import com.oceanbase.odc.service.dml.DataValue;
+
+import lombok.NonNull;
+
+/**
+ * @author jingtian
+ * @date 2023/11/22
+ * @since ODC_release_4.2.4
+ */
+public class MySQLGeometryConverter extends BaseDataConverter {
```
Comment (LuckyPickleZZ) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/dml/converter/MySQLGeometryConverter.java:31]: Could you add a unit test for this in `DataConvertUtilTest`?

Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dml/converter/MySQLGeometryConverter.java:31]: None of the other DataConverters have written ut tests, so I haven’t written ut for  MySQLGeometryConverter either. I don't think it's necessary.

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dml/converter/MySQLGeometryConverter.java:31]: DataConvertUtilTest contains all test cases for Converts

Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/dml/converter/MySQLGeometryConverter.java:31]: add ut for MySQLGeometryConverter in DataConvertUtilTest

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**